src/Simplify_Max.cpp

```diff
@@ -97,6 +97,10 @@ Expr Simplify::visit(const Max *op, ExprInfo *bounds) {
              rewrite(max(max(x, y), min(z, x)), a) ||
              rewrite(max(max(x, y), min(z, y)), a) ||
 
+             rewrite(max(select(x, max(z, y), w), z), max(select(x, y, w), z)) ||
```

Comment:
Worth adding the versions where the else case of the select is the redundant max?

Reply:
I don't think I saw these in logging, but it can't hurt. Will add